src/dart/ast.rs

```diff
@@ -557,71 +814,173 @@ impl ArgDef {
     }
 }
 
+/// Type specifier used in variable declaration.
 #[derive(Debug)]
 pub struct VarType {
+    /// The variable qualifier which comes in addition to the type.
     pub fcv: Option<FinalConstVar>,
+    /// The type of the variable to be declared (`int age = 30`).
     pub ty: Node<Type>,
 }
 
+/// Variable qualifier.
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 pub enum FinalConstVar {
+    /// A final variable can only be set once (`final name = 'Dart'`).
     Final,
+    /// A constant variable is a compile-time constant (`const length = 420`).
     Const,
+    /// Declares a variable without specifying its type (`var a = 0`).
```

Comment:
Not necessarily! The type being unspecified is represented by `Type::Infer`, we've tested and apparently having both `var` and a type name (i.e. `var Type name;`) is valid.